src/main/java/net/mcreator/ui/dialogs/wysiwyg/SpriteDialog.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2024, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.ui.dialogs.wysiwyg;
+
+import net.mcreator.blockly.data.Dependency;
+import net.mcreator.element.parts.gui.GUIComponent;
+import net.mcreator.element.parts.gui.Sprite;
+import net.mcreator.ui.component.util.PanelUtils;
+import net.mcreator.ui.help.HelpUtils;
+import net.mcreator.ui.help.IHelpContext;
+import net.mcreator.ui.init.L10N;
+import net.mcreator.ui.minecraft.TextureComboBox;
+import net.mcreator.ui.procedure.NumberProcedureSelector;
+import net.mcreator.ui.procedure.ProcedureSelector;
+import net.mcreator.ui.workspace.resources.TextureType;
+import net.mcreator.ui.wysiwyg.WYSIWYGEditor;
+import net.mcreator.workspace.elements.VariableTypeLoader;
+
+import javax.annotation.Nullable;
+import javax.swing.*;
+import java.awt.*;
+
+public class SpriteDialog extends AbstractWYSIWYGDialog<Sprite> {
+	public SpriteDialog(WYSIWYGEditor editor, @Nullable Sprite sprite) {
+		super(editor, sprite);
+		setSize(820, 170);
+		setLocationRelativeTo(editor.mcreator);
+		setModal(true);
+		setTitle(L10N.t("dialog.gui.sprite_title"));
+
+		JPanel pane = new JPanel();
+		pane.setLayout(new BoxLayout(pane, BoxLayout.PAGE_AXIS));
+
+		JPanel options = new JPanel(new BorderLayout());
+
+		TextureComboBox textureSelector = new TextureComboBox(editor.mcreator, TextureType.SCREEN, false);
+
+		JSpinner spritesCount = new JSpinner(new SpinnerNumberModel(1, 1, Integer.MAX_VALUE, 1));

```

Comment:
This parameter should not allow count above width/height of the image, or it results in zero divisions and many errors being logged

Reply:
I now pushed fix for Sprite, but limiting here may still make sense

Reply:
Fixed